include/libmpu/mpu6050.hpp

```diff
@@ -1,5 +1,217 @@
+    return hal::write(*m_i2c,
+                      m_address,
+                      std::array{ configuration_register, config[0] },
+                      hal::never_timeout());
+  }
+
+  /**
+   * @brief Re-enables acceleration readings from the MPU
+   *
+   * @return hal::status - success or errors from i2c communication
+   */
+  [[nodiscard]] hal::status power_on()
+  {
+    HAL_CHECK(is_valid_device());
+    return active_mode(true);
+  }
+
+  /**
+   * @brief Disables acceleration reading from the MPU.
+   *
+   * @return hal::status - success or errors from i2c communication
+   */
+  [[nodiscard]] hal::status power_off()
+  {
+    return active_mode(false);
+  }
+
+ private:
+  /**
+   * @brief MPU6050 Constructor
+   *
+   * @param i2c - i2c peripheral used to commnicate with device.
+   * @param address - mpu6050 device address.
+   */
+  explicit constexpr mpu6050(i2c & p_i2c, hal::byte p_device_address = 0x68)
+      : m_i2c(&p_i2c), m_address(p_device_address)
+  {
+  }
+
+  hal::result<accelerometer::read_t> driver_read() override
+  {
+    accelerometer::read_t acceleration;
+    constexpr uint16_t bytes_per_axis = 2;
+    constexpr uint8_t number_of_axis  = 3;
+
+    std::array<hal::byte, bytes_per_axis * number_of_axis> xyz_acceleration;
+    HAL_CHECK(hal::write_then_read(*m_i2c,
+                                   m_address,
+                                   std::array{ xyz_register },
+                                   xyz_acceleration,
+                                   hal::never_timeout()));
+    /**
+     * First X-axis Byte (MSB first)
+     * =========================================================================
+     * Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+     *  XD15 | XD14  |  XD13 |  XD12 |  XD11 |  XD10 |  XD9  |  XD8
+     *
+     * Final X-axis Byte (LSB)
+     * =========================================================================
+     * Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+     *   XD7 |   XD6 |   XD5 |   XD4 |   XD3 |   XD2 |   XD1 |   XD0
+     *
+     *
+     * We simply shift and OR the bytes together to get them into a signed int
+     * 16 value.
+     */
+    int16_t x =
+        static_cast<int16_t>(xyz_acceleration[0] << 8 | xyz_acceleration[1]);
+    int16_t y =
+        static_cast<int16_t>(xyz_acceleration[2] << 8 | xyz_acceleration[3]);
+    int16_t z =
+        static_cast<int16_t>(xyz_acceleration[4] << 8 | xyz_acceleration[5]);
+
+    // Convert the 16 bit value into a floating point value m/S^2
+    constexpr float kMax =
+        static_cast<float>(std::numeric_limits<int16_t>::max());
+    constexpr float kMin =
+        static_cast<float>(std::numeric_limits<int16_t>::min());
+
+    float limits       = static_cast<float>(1 << (static_cast<int>(m_gscale) + 1));
+    auto int16_bounds  = std::make_pair(kMax, kMin);
+    auto posneg_bounds = std::make_pair(-limits, limits);
```

Comment:
```suggestion
    constexpr auto input_range  = std::make_pair(max, min);
    constexpr auto output_range = std::make_pair(-limits, limits);
```

These can probably be made constexpr. Could also rename them a bit as well.